modules/sops/default.nix

```
@@ -134,35 +151,66 @@ in {
     sshKeyPaths = mkOption {
       type = types.listOf types.path;
       default = if config.services.openssh.enable then
-                  map (e: e.path) (lib.filter (e: e.type == "rsa") config.services.openssh.hostKeys)
-                else [];
+        map (e: e.path)
+        (lib.filter (e: e.type == "rsa") config.services.openssh.hostKeys)
+      else
+        [ ];
       description = ''
         Path to ssh keys added as GPG keys during sops description.
         This option must be explicitly unset if <literal>config.sops.sshKeyPaths</literal>.
       '';
     };
   };
-  config = mkIf (cfg.secrets != {}) {
-    assertions = [{
-      assertion = (cfg.gnupgHome == null) != (cfg.sshKeyPaths == []);
-      message = "Exactly one of sops.gnupgHome and sops.sshKeyPaths must be set";
-    }] ++ optionals cfg.validateSopsFiles (
-      concatLists (mapAttrsToList (name: secret: [{
-        assertion = builtins.pathExists secret.sopsFile;
-        message = "Cannot find path '${secret.sopsFile}' set in sops.secrets.${strings.escapeNixIdentifier name}.sopsFile";
-      } {
-        assertion =
-          builtins.isPath secret.sopsFile ||
-          (builtins.isString secret.sopsFile && hasPrefix builtins.storeDir secret.sopsFile);
-        message = "'${secret.sopsFile}' is not in the Nix store. Either add it to the Nix store or set sops.validateSopsFiles to false";
-      }]) cfg.secrets)
-    );
+  config = mkMerge [
+    (mkIf cfg.useAge {
```

I would like to keep the option open to support ssh later:

```suggestion
    (mkIf cfg.generateAgeKey {
```